### DISCUSSION TITLE
Mass Balance Calculation

### DISCUSSION THREAD

**Discussion by rickechler** [JSBSim-Team/jsbsim#950] [Q&A]
Hello, are the inertia products required for the entire aircraft, empty weight, including wings, tail, gear etc? Then pilot, fuel, and other point masses are calculated separately?

Thanks!

From FGMassBalance.cpp

static FGMatrix33 ReadInertiaMatrix(Element* document)
 {
   double bixx, biyy, bizz, bixy, bixz, biyz;
  
   bixx = biyy = bizz = bixy = bixz = biyz = 0.0;
   if (document->FindElement("ixx"))
     bixx = document->FindElementValueAsNumberConvertTo("ixx", "SLUG*FT2");
   if (document->FindElement("iyy"))
     biyy = document->FindElementValueAsNumberConvertTo("iyy", "SLUG*FT2");
   if (document->FindElement("izz"))
     bizz = document->FindElementValueAsNumberConvertTo("izz", "SLUG*FT2");
   if (document->FindElement("ixy"))
     bixy = document->FindElementValueAsNumberConvertTo("ixy", "SLUG*FT2");
   if (document->FindElement("ixz"))
     bixz = document->FindElementValueAsNumberConvertTo("ixz", "SLUG*FT2");
   if (document->FindElement("iyz"))
     biyz = document->FindElementValueAsNumberConvertTo("iyz", "SLUG*FT2");
  
   // Transform the inertia products from the structural frame to the body frame
   // and create the inertia matrix.
   if (document->GetAttributeValue("negated_crossproduct_inertia") == string("false"))
     return FGMatrix33( bixx,  bixy, -bixz,
                        bixy,  biyy,  biyz,
                       -bixz,  biyz,  bizz );
   else
     return FGMatrix33( bixx, -bixy,  bixz,
                       -bixy,  biyy, -biyz,
                        bixz, -biyz,  bizz );

**Comment by seanmcleod** [comment]
> Hello, are the inertia products required for the entire aircraft, empty weight, including wings, tail, gear etc? Then pilot, fuel, and other point masses are calculated separately?

Yes, take a look just a bit further down in `FGMassBalance::Run()`

https://github.com/JSBSim-Team/jsbsim/blob/e4aa2a9c873edaa8723955835e48bbcbaf195f98/src/models/FGMassBalance.cpp#L220-L236

**Reply by rickechler** [reply]
Thank you! What elements are typically included in the base configuration?

**Comment by seanmcleod** [comment]
Pretty much as you mentioned, i.e. the empty weight configuration of the aircraft.
